### PR TITLE
When handling a missing layout node, only make the placeholder present

### PR DESCRIPTION
**Pull request by ahoppen**
This fixes a couple of formatting issues and resolves the question raised in https://github.com/apple/swift-syntax/pull/1727#discussion_r1254270178.

While at it do the following
- Add a syntax trait for all missing nodes (because we’re using it for diagnostic generation)
- Improve the generation of Traits.swift
- Add missing comma in doc comments of missing nodes
- Improve `debugDescription` of `Diagnostic`
  - Since `SourceLocation` is no longer `ExpressibleByDebugDescription` due to the addition of `presumedFile` and `presumedLine`, we need to manually print just line and column here.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by bnbarham — APPROVED** [review]

**Comment by bnbarham** [comment]
> Improve the generation of Traits.swift

Are you planning on adding docs for the other traits as well?

File: Sources/SwiftParserDiagnostics/MissingNodesError.swift [hunk 1]
```diff
@@ -388,7 +388,15 @@ extension ParseDiagnosticsGenerator {
       }
     }
 
-    let changes = missingNodes.map { FixIt.MultiNodeChange.makePresent($0) }
+    let changes = missingNodes.map { node in
+      if let missing = node.asProtocol(MissingNodeSyntax.self) {
+        // For missing nodes, only make the placeholder present. Don’t make any
+        // missing nodes, e.g. in a malformed attribute, missing.
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply Sources/SwiftParserDiagnostics/MissingNodesError.swift:394]: In https://github.com/apple/swift-syntax/pull/1727/files#diff-dbdc9e7622c775bb821fe2ed1fa1ef2dd474cd2fb597480275ad048b932244c3R73-R102 we had

```swift
@_specialize(e
```

which contained a missing `:`, identifier and `)` in the attribute of the `MissingDeclSyntax`, which was made present by the Fix-It.

**Comment by ahoppen** [comment]
> Are you planning on adding docs for the other traits as well?

I am planning to add documentation to the traits as much as I’m planning to add documentation to all the syntax nodes. How soon I/the community will get to it 🤷🏽‍♂️

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**